cpp/ql/test/library-tests/dataflow/taint-tests/stl.h

```diff
@@ -185,6 +192,11 @@ namespace std
 		void str(const basic_string<charT, traits, Allocator>& str);
 	};
 
+	typedef basic_istream<char> istream;
+	typedef basic_ostream<char> ostream;
+	extern istream cin;
```

Comment:
Not really relevant for this PR, but I'm leaving a note here for future us: I would've liked to see a model for `cin` where we mark it as a `FlowSource`, but until we merge https://github.com/github/codeql/pull/3867 we can only mark it as a `RemoteFlowSource` (which probably isn't the right notion for `cin`).

Reply:
I suspect sources and sinks are one of the things still holding back our analysis and will need to become a priority before long (originally I had a test case that used `cin`, but it didn't pass and the reason didn't relate to the models being tested so I removed it).